chain/types/actor_event.go

```diff
@@ -47,9 +47,7 @@ type ActorEvent struct {
 	// Event entries in log form.
 	Entries []EventEntry `json:"entries"`
 
-	// Filecoin address of the actor that emitted this event.
-	// NOTE: In a future upgrade, this will change to always be an ID address. Currently this will be
-	// either the f4 address, or ID address if an f4 is not available for this actor.
+	// Emitter is the ID address of the actor that emitted this event.
 	Emitter address.Address `json:"emitter"`
```

Comment:
Happy to do it if there won't be any breaking changes or requirements for FIP?

Reply:
Is an API breaking change, not a networking change. Not great, but it's allowed.

Thoughts @rvagg?

Reply:
As preference I think I like `Address` here, but that's not a strong opinion. I don't like breaking APIs very much though, this would go from a JSON string to an int.

`git grep abi.ActorID -- api/` shows it being used directly in only 2 API calls, for miner worker. `git grep abi.ActorID -- chain/types` shows almost nothing. **Except** `type Event` which is essentially what we extend to make `ActorEvent` by adding tipset and message info. It's also what we return on `ChainGetEvents`, so that API is returning an int. So there's an argument for consistency there.

:shrug: I introduced `*Raw()` so we could break stuff when we do non-`Raw()` forms of them because breakage is just annoying.

How about this: I think we're going to have to rework a lot of these API guts anyway to make it tipset focused, kind of like https://github.com/filecoin-project/lotus/issues/11680 and what's been described in https://github.com/filecoin-project/lotus/issues/11770#issuecomment-2015678344. So instead of breaking this now, leave a note in here about it may break in the future to become an `ActorID`.

Reply:
@Stebalien Just to clarify, your preference is to user `ActorID` type instead of `address.Address`. Correct?

Reply:
Sorry, I meant leaving it as an `Address` (I was agreeing with rod). I'd kind of prefer ActorID from an engineering standpoint, but Address is better from a consistency standpoint.

Reply:
Got it thanks, I too think it should (eventually) be ActorID but it seems like we use address more consistently across Lotus APIs than ActorID even though in this case the address should/will always be an ID address.